features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java

```diff
@@ -168,15 +168,15 @@ public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
                 hostKeyFingerprint,
                 configType,
                 stringParameters,
-                Duration.ofMillis(timeout)
+                Duration.ofMillis(timeout).minusSeconds(1)
```

Comment:
What is the reason for subtracting one second?

Reply:
I was struggling to get any error information from the scripting service when it timed out. The monitor, the TFTP server, and the scripting service are all using the same timeout, but the monitor starts waiting first. During testing, if the script or the TFTP server hung for the entirety of the timeout, the monitor would be the first to wake up and it has no access to the internal state of the future it canceled (See [here](https://github.com/OpenNMS/opennms/blob/amay/jira/NMS-14129/features/device-config/monitor/src/main/java/org/opennms/features/deviceconfig/monitors/DeviceConfigMonitor.java#L197), the exception would always be a TimeoutException with a null message). Because of this, the REST/karaf shell commands would get a meaningless error message.
Staggering the timeouts by 1 second resolved this.